test/suite/test_search_near02.py

```diff
@@ -88,55 +104,104 @@ def test_search_near(self):
         self.assertEqual(cursor3.search_near(), -1)
         self.assertEqual(cursor3.get_key(), self.check_key("aaz"))
 
+        cursor3.set_key("aaz")
+        self.assertEqual(cursor3.search_near(), 0)
+        self.assertEqual(cursor3.get_key(), self.check_key("aaz"))
+
+        cursor3.set_key("aazab")
+        self.assertEqual(cursor3.search_near(), -1)
+        self.assertEqual(cursor3.get_key(), self.check_key("aaz"))
+
+        # Enable prefix search.
         cursor3.reconfigure("prefix_key=true")
-        # Search near for aa, should return back aaz.
+
+        # The only visible key is aaz. As long we are looking for a key that starts with either "a",
+        # "aa" or "aaz", search near should return back aaz. Otherwise, search near should return
+        # WT_NOTFOUND.
+        cursor3.set_key("a")
+        self.assertEqual(cursor3.search_near(), 1)
+        self.assertEqual(cursor3.get_key(), self.check_key("aaz"))
+
         cursor3.set_key("aa")
         self.assertEqual(cursor3.search_near(), 1)
         self.assertEqual(cursor3.get_key(), self.check_key("aaz"))
 
-        # Search near for az, should return back WT_NOTFOUND, since there are no keys
-        # visible with prefix az.
-        # FIXME-WT-8044 Prefix search near returns a valid key when expecting WT_NOTFOUND.
-        # cursor3.set_key("az")
-        # self.assertEqual(cursor3.search_near(), wiredtiger.WT_NOTFOUND)
+        cursor3.set_key("aaz")
+        self.assertEqual(cursor3.search_near(), 0)
+        self.assertEqual(cursor3.get_key(), self.check_key("aaz"))
+
+        cursor3.set_key("az")
+        self.assertEqual(cursor3.search_near(), wiredtiger.WT_NOTFOUND)
+        self.assertEqual(cursor3.get_key(), self.check_key("az"))
```

Comment:
I think in the case where we return back WT_NOTFOUND, it's not necessary to check the returned key.

Reply:
Fixed in 0d04e731feba5ae0c6b40c4005ac0e71574e7391
Sorry for the commit message, I don't know how I ended up writing it this way...